frontend/app/src/utils/history/events.ts

```diff
@@ -0,0 +1,32 @@
+import { type MaybeRef } from '@vueuse/core';
+import { HistoryEventEntryType } from '@rotki/common/lib/history/events';
+
+export const isEvmEvent = (type: MaybeRef<HistoryEventEntryType>) =>
+  get(type) === HistoryEventEntryType.EVM_EVENT;
+
+export const isEvmEventRef = (
+  type: MaybeRef<HistoryEventEntryType>
+): ComputedRef<boolean> => computed(() => isEvmEvent(type));
```

Comment:
To resolve the casting issue just change the first to a type guard:

```typescript
export const isEvmEvent = (
  type: MaybeRef<HistoryEventEntryType>
): type is MaybeRef<EvmHistoryEvent> =>
  get(type) === HistoryEventEntryType.EVM_EVENT;
```

And make the second to return either object if the type guard is true or `undefined` if false so that isEvmEventRef returns `ComputedRef<EvmHistoryEvent | undefined>` you can do a v-if on the undefined/defined and narrow the types in the checks and you don't have to cast or lose types.

Reply:
Oh yeah this is a cool idea, will try it